great_expectations/datasource/fluent/data_asset/data_connector/azure_blob_storage_data_connector.py

```diff
@@ -122,6 +128,7 @@ def build_data_connector(  # noqa: PLR0913
             container: container name for Microsoft Azure Blob Storage
             name_starts_with: Microsoft Azure Blob Storage prefix
             delimiter: Microsoft Azure Blob Storage delimiter
+            recursive_file_discovery: Flag to indicate if files should be searched recursively from subfolders
```

Comment:
@toivomattila This docstring addition should also go to the top of the module (both).  Thanks.